src/webrtc/gcc.md

```diff
@@ -0,0 +1,159 @@
+packets will be delayed and finally dropped. Such situation is called the network congestion.
+
+Congestion control goal is to prevent from congesting the network by controling the sender rate.
+Every time, the sender notices that the packet transmission time increases or packets are getting
+lost it should lower its rate.
+
+<!-- In the case of multimedia, lowering sender rate is challenging as the audio and video 
+have specified bitrate and sometimes even lowering the video quality might not  -->
+
+## TWCC vs GCC
+
+TWCC (Transport-wide Congestion Control) defines new RTP header extension and RTCP message used to 
+feedback the arrival times of RTP packets to the sender.
+Each RTP packet is assigned transport-wide (unique across all RTP streams sent at the same socket) 
+sequence number by the sender before transmission.
+The receiver records arrival timestamps of packets and send them back in TWCC Feedback message.
+Then, the sender is able to determine network delay and packet loss, and react to them by lowering its rate.
+How this reaction looks like, i.e. how much sender should lower its rate, when it should increase and when 
+it should hold its rate, etc. is out-of-scope of TWCC and depends on specific congestion control algorithm.
+In other words, TWCC provides only means to implement actual congestion control algorithm.
+
+GCC (Google Congestion Control) is actual congestion control algorithm developed by Google.
+It uses TWCC Feedback messages to determine when and how much to increase and decrease sender rate, 
+or when the sender rate should remain at the current level.
+
+## GCC
+
+GCC goal is to estimate the rate that sender can send its data with.
+The whole system depends on over-using the channel i.e. to get higher estimation we have to increase our rate.
+From the RFC draft:
+
+> the system depends on over-using the channel to verify the current available bandwidth estimate, we must make
+> sure that our estimate does not diverge from the rate at which the sender is actually sending.  
+> Thus, if the sender is unable to produce a bit stream with the bitrate the congestion controller is asking for, 
+> the available bandwidth estimate should stay within a given bound.
+
+The bandwidth estimation provided by congestion controller is in fact a request to the sender for producing 
+the stream with specific bitrate.
+If the network behavior remains unchanged after increasing sender rate the estimation is increased.
+If the network starts being congested (transmission delay increases, packets get lost), the estimation
+is decreased.
+This way we are slowly increasing the sender rate up to the point where the network can't accept more data.
+
+There are two increase modes: multiplicative and additive.
+In the multiplicative mode, the estimation is increase max by 8% per second.
+In the additvie mode, the estimation is increased with at most half a packet per `response_time_interval` (RTT + 100ms).
```

Comment:
I'm not sure how to interpret the second part of the sentence. But I think I would write similarly to the previous one.
```suggestion
In the additive mode, the estimation can be increased by no more than half a packet per `response_time_interval` (RTT + 100ms).
```

Reply:
I took it from the RFC :roll_eyes:

Reply:
There might be room for improvement over the RFC :roll_eyes: